command/certificate/verify.go

```diff
@@ -180,5 +247,129 @@ func verifyAction(ctx *cli.Context) error {
 		return errors.Wrapf(err, "failed to verify certificate")
 	}
 
+	verboseMSG := "certificate validated against roots\n"
+	if host != "" {
+		verboseMSG += "certificate host name validated\n"
+	}
+
+	if issuerFile != "" {
+		issuerCertPEM, err := os.ReadFile(issuerFile)
+		if err != nil {
+			return errors.Errorf("unable to load the issuing CA certificate file")
+		}
+
+		issuerBlock, _ := pem.Decode(issuerCertPEM)
+		if issuerBlock == nil || issuerBlock.Type != "CERTIFICATE" {
+			return errors.Errorf("failed to decode the issuing CA certificate")
+		}
+
+		issuer, err = x509.ParseCertificate(issuerBlock.Bytes)
+		if err != nil {
+			return errors.Errorf("failed to parse the issuing CA certificate")
+		}
+	}
+
+	if verifyCRL {
+		var endpoint string
+		switch {
+		case crlEndpoint != "" && issuer.Equal(&x509.Certificate{}):
+			return errors.Errorf("you must provider the issuer CA if you are going to specificy the CRL endpoint")
+		case crlEndpoint != "":
+			endpoint = crlEndpoint
+		case len(cert.CRLDistributionPoints) == 0:
+			return errors.Errorf("CRL distribution endpoint not found in certificate")
+		default:
+			endpoint = cert.CRLDistributionPoints[0]
+		}
+
+		httpClient := http.Client{}
+
+		resp, err := httpClient.Get(endpoint)
+		if err != nil {
+			return errors.Wrap(err, "error downloading crl")
+		}
+		defer resp.Body.Close()
+		if resp.StatusCode >= 400 {
+			return errors.Errorf("error downloading crl: status code %d", resp.StatusCode)
+		}
+		b, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return errors.Wrap(err, "error downloading crl")
+		}
+
+		crl, err := x509.ParseRevocationList(b)
+		if err != nil {
+			return errors.Wrap(err, "error parsing crl")
+		}
+
+		crlJSON, err := crlutil.ParseCRL(b)
+		if err != nil {
+			return errors.Wrap(err, "error parsing crl into json")
+		}
+
+		if !issuer.Equal(&x509.Certificate{}) && !insecure {
```

Comment:
I think this should be:
```suggestion
		if issuer != nil && !insecure {
```